Fix/stuff

rndmh3ro opened the pull request:
@moritzrp wanna take a look? :)

moritzrp commented:
Looks good from my side! Only a small thing from my side: Since the description can be a list as well, I would add it to the annotation of RoleDefault. So `description: Union[str, list[str]]` instead of just `description: str`. I missed that part.

schurzi reviewed (approved):
There are also some references to PyYAML in `.pre-commit-config.yaml`, should the be removed too?

rndmh3ro commented:
@moritzrp, thanks for noticing! Fixed this.

@schurzi, thanks for this. I noticed that there was still pyyaml used in the core-functions. I removed it there, too and replaced it with ruamel.yaml. I also changed the handling of defaults and made it hopefully simpler.

Ready to review now.
